src/xcode/ENA/ENA/Source/Scenes/Onboarding/OnboardingInfoViewController.swift

```diff
@@ -226,10 +301,19 @@ final class OnboardingInfoViewController: UIViewController {
 		stateTitleLabel.text = onboardingInfo.stateTitle
 		stateStateLabel.text = exposureNotificationsEnabled ? onboardingInfo.stateActivated : onboardingInfo.stateDeactivated
 
-		guard !pageSetupDone else {
+		if pageType == .enableLoggingOfContactsPage && showStateView {
+			nextButton.setTitle(onboardingInfo.alternativeActionText, for: .normal)
+		}
```

Comment:
I think Artur initially suggested to combine this with the original `nextButton.setTitle(onboardingInfo.actionText, for: .normal)` and I like that suggestion. You could move it up before line `nextButton.isHidden = onboardingInfo.actionText.isEmpty` and write it like this:

```
if pageType == .enableLoggingOfContactsPage && showStateView {
    nextButton.setTitle(onboardingInfo.alternativeActionText, for: .normal)
} else {
    nextButton.setTitle(onboardingInfo.actionText, for: .normal)
}
```

Reply:
You're right, this avoids confusion.